### PR TITLE
[MODORGSTOR-153] - BE - Implement API for Privileged contacts

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://issues.folio.org/browse/MODORGSTOR-153

## Approach
Created corresponding table and API.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate action.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code are 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added, or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -272,7 +303,7 @@
     },
     {
       "id": "organizations-storage.banking-account-types",
-      "version": "1.0",
+      "version": "4.7",
```
Comment (imerabishvili) [inline descriptors/ModuleDescriptor-template.json:306]: Is this correct?

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -95,6 +95,37 @@
         }
       ]
     },
+    {
+      "id": "organizations-storage.privileged-contacts",
+      "version": "4.7",
```
Comment (SerhiiNosko) [inline descriptors/ModuleDescriptor-template.json:100]: version should be 1.0
It is version of API, not module

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (SerhiiNosko) [reply descriptors/ModuleDescriptor-template.json:306]: no, it should be 1.0

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (imerabishvili) [reply descriptors/ModuleDescriptor-template.json:100]: and why do we have 2.0 for contacts, what's the difference?

File: src/main/resources/templates/db_scripts/schema.json [hunk 3]
```diff
@@ -185,6 +191,11 @@
       "tableName": "settings",
       "fromModuleVersion": "mod-organizations-storage-4.7.0",
       "withMetadata": true
+    },
+    {
+      "tableName": "privileged_contacts",
+      "fromModuleVersion": "mod-organizations-storage-4.7.0",
+      "withMetadata": true
```
Comment (SerhiiNosko) [inline src/main/resources/templates/db_scripts/schema.json:198]: We have some tickets in backlog to enable optimistic locking for existing tables like this: https://issues.folio.org/browse/MODORGSTOR-106
I think we also should enable it for new tables.
I think we need to do the same for organization_types table

Comment (SerhiiNosko) [reply src/main/resources/templates/db_scripts/schema.json:198]: Lets check in architects do we need to enable optimistic locking for new tables

File: src/main/java/org/folio/rest/impl/TenantReferenceAPI.java [hunk 4]
```diff
@@ -66,6 +66,11 @@ private void buildDataLoadingParameters(TenantAttributes tenantAttributes, Tenan
         .withLead("data")
         .add("interfaces-2.0.0", "organizations-storage/interfaces");
       }
+      if (isNew(tenantAttributes, "4.7.0")) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/TenantReferenceAPI.java:69]: We need to re-check in our POs do we need to add reference data for priviledge contacts

Comment (imerabishvili) [reply src/main/java/org/folio/rest/impl/TenantReferenceAPI.java:69]: I checked and we don't need them

File: src/main/resources/templates/db_scripts/schema.json (hunk 3 above)
Comment (imerabishvili) [reply src/main/resources/templates/db_scripts/schema.json:202]: enabled for new tables

**Review by SerhiiNosko — APPROVED** [review]

**Review by Dmitriy-Butramyou — APPROVED** [review]

**Pull request merged**